asset/asset.go

```diff
@@ -939,18 +939,26 @@ func New(genesis Genesis, amount, locktime, relativeLocktime uint64,
 			genesis.Type)
 	}
 
+	// Valid genesis asset witness.
+	genesisWitness := Witness{
+		PrevID:          &PrevID{},
+		TxWitness:       nil,
+		SplitCommitment: nil,
+	}
+
+	// Genesis assets with an asset group must have the group witness stored
+	// in the genesis asset witness, if present.
+	if groupKey != nil && groupKey.Witness != nil {
```

Comment:
Still think we don't necessarily need the witness field within the `GroupKey` itself. I can see how it's convenient here though since we're already accepting in the group key.

Reply:
Also, should we use `len(groupKey.Witness) != 0` to avoid this being true for empty witnesses?

Reply:
Should use length here, otherwise this should get rejected later as it has a witness, but may not be prev out zero or nl,e tc.